Source/Core/DolphinTool/VerifyCommand.cpp

```diff
@@ -0,0 +1,180 @@
+    return 1;
+  }
+
+  // Open the volume
+  std::shared_ptr<DiscIO::VolumeDisc> volume = DiscIO::CreateDisc(input_file_path);
+  if (!volume)
+  {
+    std::cerr << "Error: Unable to open disc image" << std::endl;
+    return 1;
+  }
+
+  // Verify the volume
+  const std::optional<DiscIO::VolumeVerifier::Result> result =
+      VerifyVolume(volume, enable_crc32, enable_md5, enable_sha1);
+  if (!result)
+  {
+    std::cerr << "Error: Unable to verify volume" << std::endl;
+    return 1;
+  }
+
+  if (algorithm == std::nullopt)
+  {
+    PrintFullReport(result);
+  }
+  else
+  {
+    if (enable_crc32 && !result->hashes.crc32.empty())
+      std::cout << HashToHexString(result->hashes.crc32) << std::endl;
+    else if (enable_md5 && !result->hashes.md5.empty())
+      std::cout << HashToHexString(result->hashes.md5) << std::endl;
+    else if (enable_sha1 && !result->hashes.sha1.empty())
+      std::cout << HashToHexString(result->hashes.sha1) << std::endl;
+    else
+    {
+      std::cerr << "Error: No hash available" << std::endl;
+      return 1;
+    }
+  }
+
+  return 0;
+}
+
+void VerifyCommand::PrintFullReport(const std::optional<DiscIO::VolumeVerifier::Result> result)
+{
+  if (!result->hashes.crc32.empty())
+    std::cout << "CRC32: " << HashToHexString(result->hashes.crc32) << std::endl;
+  else
+    std::cout << "CRC32 not available" << std::endl;
+
+  if (!result->hashes.md5.empty())
+    std::cout << "MD5: " << HashToHexString(result->hashes.md5) << std::endl;
+  else
+    std::cout << "MD5 not available" << std::endl;
+
+  if (!result->hashes.sha1.empty())
+    std::cout << "SHA1: " << HashToHexString(result->hashes.sha1) << std::endl;
+  else
+    std::cout << "SHA1 not available" << std::endl;
+
+  std::cout << "Problems Found: " << (result->problems.size() > 0 ? "Yes" : "No") << std::endl;
+
+  for (int i = 0; i < static_cast<int>(result->problems.size()); ++i)
+  {
+    const DiscIO::VolumeVerifier::Problem problem = result->problems[i];
+
+    std::cout << std::endl << "Severity: ";
+    switch (problem.severity)
+    {
+    case DiscIO::VolumeVerifier::Severity::Low:
+      std::cout << "Low";
+      break;
+    case DiscIO::VolumeVerifier::Severity::Medium:
+      std::cout << "Medium";
+      break;
+    case DiscIO::VolumeVerifier::Severity::High:
+      std::cout << "High";
+      break;
+    case DiscIO::VolumeVerifier::Severity::None:
+      std::cout << "None";
+      break;
+    default:
+      ASSERT(false);
+      break;
+    }
+    std::cout << std::endl;
+
+    std::cout << "Summary: " << problem.text << std::endl << std::endl;
+  }
+}
+
+std::optional<DiscIO::VolumeVerifier::Result>
+VerifyCommand::VerifyVolume(std::shared_ptr<DiscIO::VolumeDisc> volume, bool enable_crc32,
+                            bool enable_md5, bool enable_sha1)
+{
+  if (!volume)
+    return std::nullopt;
+
+  DiscIO::VolumeVerifier verifier(*volume, false, {enable_crc32, enable_md5, enable_sha1});
```

Comment:
Do we not want to provide an option for Redump verification?

Reply:
I did not implement this, at least for V1 of this project, for 2 reasons 
1 - Didn't want this to be accidentally used to abuse Redump server in a batch setting
2 - Had issues setting up a temp directory to cache the DAT, ended up being a rabbit hole for now

Reply:
Good point. Let's leave it as it is for now.